install.sh

```diff
@@ -1,23 +1,35 @@
 #!/bin/bash
 
-# This script will create links for the dotfiles.
-
-##########
-#  nvim  #
-##########
-mkdir -p "$HOME/.config/nvim"
-mkdir -p "$HOME/.config/nvim/undo"
-ln -sf "$HOME/dotfiles/nvim/init.vim" "$HOME/.config/nvim/init.vim"
+# TODO(collins): add ~/.config/nvim/init.vim to dotfiles/ in this git repo
+# TODO(collins): add ~/.config/X11 to dotfiles/ in this git repo
+# TODO(collins): add ~/.vim to dotfiles/ in this git repo
+# TODO(collins): add ~/.vimrc to dotfiles/ in this git repo
 
+# This script will create links for the dotfiles.
 
-##############
-# Xresources #
-##############
-ln -sfn "$HOME/dotfiles/X11" "$HOME/.config"
-
+# Step 1: ensure all necessary directories exist before linking files.
+for dir in $(find dotfiles/ -mindepth 1 -type d); do
+	target="$HOME/${dir#dotfiles/}"
+	echo "INFO: Ensuring directory exists: $target"
+	mkdir -p $target
+done
 
-########
-#  vim #
-########
-ln -s "$HOME/dotfiles/vim/.vim" "$HOME/.vim"
-ln -s "$HOME/dotfiles/vim/.vimrc" "$HOME/.vimrc"
+# Step 2: create links for all dotfiles
+for file in $(find dotfiles/ -mindepth 1 -type f); do
+	source=$PWD/$file
+	target="$HOME/${file#dotfiles/}"
+	# This if statement could be replaced by passing -f to ln,
```

Comment:
Yes, and you can use it if you want, but as I say in the comment, it's usually better to warn instead of force, so that you don't accidentally end up deleting a file (remember that `-f` will delete not just an existing symbolic link, but also a regular file, because a regular file is just a hard link to an inode).